romancal/pipeline/exposure_pipeline.py

```diff
@@ -71,66 +75,95 @@ def process(self, input):
+                log.debug("Error opening file:")
+
+        # Build a list of observations to process
+        expos_file = []
+        if file_type == "asdf":
+            expos_file = [input]
+        elif file_type == "asn":
+            for product in asn["products"]:
+                for member in product["members"]:
+                    expos_file.append(member["expname"])
+
+        results = []
+        for in_file in expos_file:
+            if isinstance(in_file, str):
+                input_filename = basename(in_file)
+                log.info(f"Input file name: {input_filename}")
+            else:
+                input_filename = None
+
+            # Open the file
+            input = rdm.open(in_file)
+            log.info(f"Processing a WFI exposure {in_file}")
+
+            self.dq_init.suffix = "dq_init"
+            result = self.dq_init(input)
+            if input_filename:
+                result.meta.filename = input_filename
+            result = self.saturation(result)
+
+            # Test for fully saturated data
             if is_fully_saturated(result):
-                # Set all subsequent steps to skipped
-                for step_str in [
-                    "assign_wcs",
-                    "flat_field",
-                    "photom",
-                    "source_detection",
-                ]:
-                    result.meta.cal_step[step_str] = "SKIPPED"
-
-                # Set suffix for proper output naming
-                self.suffix = "cal"
+                log.info("All pixels are saturated. Returning a zeroed-out image.")
+
+                # Return zeroed-out image file (stopping pipeline)
+                return self.create_fully_saturated_zeroed_image(result)
+
+            result = self.linearity(result)
+            result = self.dark_current(result)
+            result = self.jump(result)
+            result = self.rampfit(result)
+
+            # Test for fully saturated data
+            if "groupdq" in result.keys():
+                if is_fully_saturated(result):
+                    # Set all subsequent steps to skipped
+                    for step_str in [
+                        "assign_wcs",
+                        "flat_field",
+                        "photom",
+                        "source_detection",
+                    ]:
+                        result.meta.cal_step[step_str] = "SKIPPED"
+
+                    # Set suffix for proper output naming
+                    self.suffix = "cal"
+                    results.append(result)
 
                 # Return fully saturated image file (stopping pipeline)
-                return result
-
-        result = self.assign_wcs(result)
-        if result.meta.exposure.type == "WFI_IMAGE":
-            result = self.flatfield(result)
-        else:
-            log.info("Flat Field step is being SKIPPED")
-            result.meta.cal_step.flat_field = "SKIPPED"
+                return results
```

Comment:
I'm confused about this.  The old version had this line under the is_fully_saturated condition; here this is outside that condition, so if there is ' groupdq' in result.keys() we stop the pipeline?  I think I would have guessed that this should be indented one block, and that rather than returning, we continue to the next SCA?

Reply:
That looks like a rebase/merge issue and I'll fix that.